src/tests/PostTest.test.ts

```diff
@@ -0,0 +1,318 @@
+    const postController = ControllerFactory.post(conn);
+    const post = PostFactory.fakeTemplate();
+    post.user = UserFactory.fakeTemplate();
+
+    await new DataFactory()
+      .createPosts(post)
+      .createUsers(post.user)
+      .write();
+
+    expectedPost.user = post.user;
+
+    const search = {
+      keywords: 'Kombucha',
+    }
+
+    const getPostsResponse = await postController.searchPosts(search);
+    getPostsResponse.posts[0].price = Number(getPostsResponse.posts[0].price);
+
+    expect(getPostsResponse.posts).toEqual([expectedPost]);
+  });
+
+  test('search posts - substring', async () => {
+    const postController = ControllerFactory.post(conn);
+    const post = PostFactory.fakeTemplate();
+    post.user = UserFactory.fakeTemplate();
+
+    await new DataFactory()
+      .createPosts(post)
+      .createUsers(post.user)
+      .write();
+
+    expectedPost.user = post.user;
+    
+    const search = {
+      keywords: 'S KOM',
+    }
+
+    const getPostsResponse = await postController.searchPosts(search);
+    getPostsResponse.posts[0].price = Number(getPostsResponse.posts[0].price);
+    
+    expect(getPostsResponse.posts).toEqual([expectedPost]);
+  });
+
+  test('search posts - case mismatch', async () => {
+    const postController = ControllerFactory.post(conn);
+    const post = PostFactory.fakeTemplate();
+    post.user = UserFactory.fakeTemplate();
+
+    await new DataFactory()
+      .createPosts(post)
+      .createUsers(post.user)
+      .write();
+
+    expectedPost.user = post.user;
+
+    const search = {
+      keywords: 'FermenteD',
+    }
+
+    const getPostsResponse = await postController.searchPosts(search);
+    getPostsResponse.posts[0].price = Number(getPostsResponse.posts[0].price);
+    
+    expect(getPostsResponse.posts).toEqual([expectedPost]);
+  });
+
+  test('search posts - no matches', async () => {
+    const postController = ControllerFactory.post(conn);
+    const post = PostFactory.fakeTemplate();
+    post.user = UserFactory.fakeTemplate();
+
+    await new DataFactory()
+      .createPosts(post)
+      .createUsers(post.user)
+      .write();
+
+    expectedPost.user = post.user;
+
+    const search = {
+      keywords: 'Kobucha',
+    }
+
+    const getPostsResponse = await postController.searchPosts(search);
+    
+    expect(getPostsResponse.posts).toEqual([]);
+  });
+
+  test('filter posts', async () => {
+    const postController = ControllerFactory.post(conn);
+    const post = PostFactory.fakeTemplate();
+    post.user = UserFactory.fakeTemplate();
+
+    await new DataFactory()
+      .createPosts(post)
+      .createUsers(post.user)
+      .write();
+      
+    expectedPost.user = post.user;
+
+    let filter = {
+      category: 'HANDMADE',
+    }
+
+    let getPostsResponse = await postController.filterPosts(filter);
+    getPostsResponse.posts[0].price = Number(getPostsResponse.posts[0].price);
+
+    expect(getPostsResponse.posts).toEqual([expectedPost]);
```

Comment:
ideally we have a single expect in each test case unless they're related - here I see test cases one where the filter returns something and another where it doesn't which are two unique test cases and here you are also testing if some post that has 2 possible filters is detected by both which is another